File: Data Structures & Algorithms/valid-sudoku/submission-0.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        row_set = collections.defaultdict(set)
        col_set = collections.defaultdict(set)
        square = collections.defaultdict(set)


        for row in range(len(board)):
            for col in range(len(board[0])):
                if board[row][col] == '.':
                    continue
                if board[row][col] in row_set[row] or board[row][col] in col_set[col] or board[row][col] in square[(row//3, col//3)]:
                    return False
                row_set[row].add(board[row][col])
                col_set[col].add(board[row][col])
                square[(row//3, col//3)].add(board[row][col])
        return True

File: Data Structures & Algorithms/count-number-of-islands/submission-0.java
class Solution {
    public int numIslands(char[][] grid) {
        int numOfIslands = 0;
        for(int row = 0; row < grid.length; row ++) {
            for(int col = 0; col < grid[0].length; col ++) {
                if(grid[row][col] == '1') {
                    numOfIslands++;
                    dfs(row, col, grid);
                }
                
            }
        }
        return numOfIslands; 
    }

    public void dfs(int row, int col, char[][] grid) {
        if(row < 0 || col < 0 || row >= grid.length || col >= grid[0].length || grid[row][col] != '1') {
            return;          
        }
        grid[row][col] = '#';
        dfs(row, col + 1, grid);
        dfs(row, col - 1, grid);
        dfs(row + 1, col, grid);
        dfs(row - 1, col, grid);
    }
}
